File: invesdwin-scripting-parent/invesdwin-scripting-rust/invesdwin-scripting-rust-runtime-contract/src/test/java/de/invesdwin/scripting/rust/runtime/contract/callback/ParametersAndReturnsTestDouble.rs
println!("getDouble");
let getDouble: f64 = callback("getDouble", &[]);
println!("getDouble: {}", getDouble);
callback_void("setDouble", &[param(getDouble)]);

println!("getDoubleVector");
let getDoubleVector: Vec<f64> = callback("getDoubleVector", &[]);
println!("getDoubleVector: {:?}", getDoubleVector);
callback_void("setDoubleVector", &[param(getDoubleVector)]);

println!("getDoubleVectorAsList");
let getDoubleVectorAsList: Vec<f64> = callback("getDoubleVectorAsList", &[]);
println!("getDoubleVectorAsList: {:?}", getDoubleVectorAsList);
callback_void("setDoubleVectorAsList", &[param(getDoubleVectorAsList)]);

println!("getDoubleMatrix");
let getDoubleMatrix: Vec<Vec<f64>> = callback("getDoubleMatrix", &[]);
println!("getDoubleMatrix: {:?}", getDoubleMatrix);
callback_void("setDoubleMatrix", &[param(getDoubleMatrix)]);

println!("getDoubleMatrixAsList");
let getDoubleMatrixAsList: Vec<Vec<f64>> = callback("getDoubleMatrixAsList", &[]);
println!("getDoubleMatrixAsList: {:?}", getDoubleMatrixAsList);
callback_void("setDoubleMatrixAsList", &[param(getDoubleMatrixAsList)]);

File: invesdwin-scripting-parent/invesdwin-scripting-haskell/invesdwin-scripting-haskell-runtime-ghci/src/test/java/de/invesdwin/scripting/haskell/runtime/callback/ParametersAndReturnsTestDouble.hs
putStrLn ( show ( "getDouble" ) )
getDouble :: Double <- callback "getDouble" []
putStrLn ( show ( getDouble ) )
callback "setDouble" [param getDouble] :: IO ()

putStrLn ( show ( "getDoubleVector" ) )
getDoubleVector :: [Double] <- callback "getDoubleVector" []
putStrLn ( show ( getDoubleVector ) )
callback "setDoubleVector" [param getDoubleVector] :: IO ()

putStrLn ( show ( "getDoubleVectorAsList" ) )
getDoubleVectorAsList :: [Double] <- callback "getDoubleVectorAsList" []
putStrLn ( show ( getDoubleVectorAsList ) )
callback "setDoubleVectorAsList" [param getDoubleVectorAsList] :: IO ()

putStrLn ( show ( "getDoubleMatrix" ) )
getDoubleMatrix :: [[Double]] <- callback "getDoubleMatrix" []
putStrLn ( show ( getDoubleMatrix ) )
callback "setDoubleMatrix" [param getDoubleMatrix] :: IO ()

putStrLn ( show ( "getDoubleMatrixAsList" ) )
getDoubleMatrixAsList :: [[Double]] <- callback "getDoubleMatrixAsList" []
putStrLn ( show ( getDoubleMatrixAsList ) )
callback "setDoubleMatrixAsList" [param getDoubleMatrixAsList] :: IO ()
